src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/VariantsSparkSink.java

```diff
@@ -178,6 +178,12 @@ private static void writeVariantsSingle(
             final VCFHeader header, final boolean writeGvcf, final List<Integer> gqPartitions, final int defaultPloidy, final int numReducers) throws IOException {
 
         final Configuration conf = ctx.hadoopConfiguration();
+
+        //TODO remove me when https://github.com/broadinstitute/gatk/issues/4274 is fixed
+        if (writeGvcf && (AbstractFeatureReader.hasBlockCompressedExtension(outputFile) || outputFile.endsWith(IOUtil.BCF_FILE_EXTENSION))) {
+            throw new UserException.UnimplementedFeature("It is currently not possible to write a compressed g.vcf or bcf.gz on spark.  See https://github.com/broadinstitute/gatk/issues/4274 for more details.");
```

Comment:
Doesn't this forbid writing any BCFs when in GVCF mode (not just `bcf.gz`)? Was this the intent? Both the error message and the ticket https://github.com/broadinstitute/gatk/issues/4274 seem incomplete, since they don't mention disallowing `g.bcf`

Reply:
That was the intent, it's currently not possible to write g.bcf at all, compressed or not.  I think we are actually compressing bcf though automatically on spark, but it interacts strangely with the .gz codepaths.  I've updated the comment and opened a new issue https://github.com/broadinstitute/gatk/issues/4303